### PR TITLE
schedule: fix the down peer cannot be repaired

### LINKED ISSUE #7808: Down-peer-region can't recover when enable placement-rule policy
### What version of TiKV are you using?

v6.5.3
### Steps to reproduce

1. set placement-rule:
```
[
  {
    "group_id": "pd",
    "id": "1",
    "start_key": "",
    "end_key": "",
    "role": "voter",
    "is_witness": false,
    "count": 2,
    "label_constraints": [
      {
        "key": "disk_type",
        "op": "in",
        "values": [
          "ssd"
        ]
      }
    ],
    "location_labels": [
      "host"
    ],
    "isolation_level": "host",
    "create_timestamp": 1706696121
  },
  {
    "group_id": "pd",
    "id": "2",
    "start_key": "",
    "end_key": "",
    "role": "follower",
    "is_witness": false,
    "count": 1,
    "label_constraints": [
      {
        "key": "disk_type",
        "op": "in",
        "values": [
          "mix"
        ]
      }
    ],
    "location_labels": [
      "host"
    ],
    "isolation_level": "host",
    "create_timestamp": 1706696121
  }
]
```
2. Start 8 tikv nodes, where 3 nodes are labeled `disk_type=mix`, and the other 5 nodes are labeled `disk_type=ssd`. Then load data and observe that there is no leader on the `mix` node, which is expected.
image
3. force scale-in one `mix` node, find there are lots of `down-peer` not recover
image
4. disable placement-rule, then `down-peer` starts to recover

Is this caused by my wrong placement policy or a bug of placement-rule?

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #7808.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/checker/replica_strategy.go [hunk 1]
```diff
@@ -98,7 +98,12 @@ func (s *ReplicaStrategy) SelectStoreToFix(coLocationStores []*core.StoreInfo, o
 	}
 	// trick to avoid creating a slice with `old` removed.
 	s.swapStoreToFirst(coLocationStores, old)
-	return s.SelectStoreToAdd(coLocationStores[1:])
+	// If the coLocationStores only has one store, no need to remove.
```
Comment (CabinfeverB) [inline pkg/schedule/checker/replica_strategy.go:101]: In issue desc, there are 3 nodes labeled `disk_type=mix`.  Does it affect 3 nodes case?

**Review by CabinfeverB — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Comment by Smityz** [comment]
Thank you! @rleungx

**Comment by rleungx** [comment]
> Thank you! @rleungx

Can you try it again?

**Comment by Smityz** [comment]
> > Thank you! @rleungx
> 
> Can you try it again?

I will try later

**Pull request merged**